nixos/modules/services/desktops/pipewire.nix

```diff
@@ -17,6 +17,26 @@ let
     mkdir -p "$out/lib"
     ln -s "${cfg.package.jack}/lib" "$out/lib/pipewire"
   '';
+
+  # Helpers for generating the pipewire JSON config file
+  mkSPAValueString = v:
+    if      builtins.isList       v then "[${lib.concatMapStringsSep " " mkSPAValueString v}]"
+    else if lib.types.attrs.check v then
+      "{${lib.concatStringsSep " " (mkSPAKeyValue { _SPA_DUP = v._type or "" == "_SPA_DUP"; } v)}}"
+    else lib.generators.mkValueStringDefault {} v;
+
+  mkSPAKeyValue = { _SPA_DUP ? false }: attrs: map (def: def.content) (lib.sortProperties
```

Comment:
I couldn't get this working in any sort of elegant fashion, though the only part affected by this is the `objects` field for `pipewire.conf`.

Reply:
The simplified version that just gets rid of the `_SPA_DUP`

```nix
  mkSPAKeyValue = attrs: map (def: def.content) (
    lib.sortProperties
      (
        lib.mapAttrsToList
          (k: v: lib.mkOrder (v._priority or 1000) "${lib.escape [ "=" ] k} = ${mkSPAValueString (v._content or v)}")
          attrs
      )
  );
```

Reply:
Hmm how is that meant to be used? With the current `toSPAJSON = attrs: lib.concatStringsSep "\n" (mkSPAKeyValue {} attrs);` these just cause a "attempt to call something which is not a function but a list.

Reply:
How about this?

Reply:
If we're making converting the JSON to Nix sets before reconverting that to SPA JSON then more or less you'll still need parts of this. `sortProperties` will definitely be needed because of the `load-module` ordering dependency.

Reply:
Also get's messy if there are multiple `objects` with the same factory since duplicate keys aren't allowed in Nix sets.